### PR TITLE
Add new test script for functional tests, run on Jenkins

### PR DESCRIPTION
**Pull request by rmohr**
Since all jenkins nodes already have vagrant, libvirt and libvirt-devel
installed, this test script only installs go dependencies, tries to
provision the Vagrant VMs and run the tests against it. All other files
in the automation folder are now obsolete.

Signed-off-by: Roman Mohr <rmohr@redhat.com>

### CODE REVIEW

**Comment by rmohr** [comment]
Don't merge, until I enabled Jenkins, and this PR shows that it works.

File: automation/test.sh [hunk 1]
```diff
@@ -55,10 +46,10 @@ vagrant up --provider=libvirt
 if [ $? -ne 0 ]; then
   # After a workspace cleanup we loose our .vagrant file, this means that we have to clean up libvirt
   vagrant destroy
-  virsh undefine kubevirt_master
   virsh destroy kubevirt_master
-  virsh undefine kubevirt_node0
+  virsh undefine kubevirt_master
   virsh destroy kubevirt_node0
+  virsh undefine kubevirt_node0
```
Comment (davidvossel) [inline automation/test.sh]: It is possible you may find you have to manually remove the storage volumes too here if you lost the .vagrant file.   If you've tested this with multiple runs and it works, then the volumes probably aren't an issue.

**Comment by rmohr** [comment]
retest this please

**Comment by stu-gott** [comment]
LGTM

**Comment by rmohr** [comment]
retest this please

**Pull request merged**